.github/workflows/DrainpipeDev.yml

```diff
@@ -1,50 +1,60 @@
-name: "Push Drainpipe Dev Package"
+name: Push Drainpipe Dev Package
 
 on:
   push:
     branches:
+      - 460-drainpipe-dev-use-deploy-token
       - main
     tags:
       - 'v*'
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  group: ${{ github.workflow }}-${{ github.ref_name || github.sha }}
   cancel-in-progress: false
 
+permissions:
+  contents: read
+
 jobs:
   Drainpipe-Dev:
-    runs-on: ubuntu-22.04
+    name: Push branch to drainpipe-dev
+    runs-on: ubuntu-latest
     steps:
-
-      - name: Create a Drupal project
-        run: composer create-project drupal/recommended-project . --ignore-platform-req=ext-gd
-
       - uses: actions/checkout@v4
-        with:
-          path: drainpipe
 
-      - uses: ./drainpipe/scaffold/github/actions/common/set-env
-
-      - name: Install DDEV
-        uses: ./drainpipe/scaffold/github/actions/common/ddev
+      - name: Add drainpipe-dev deploy key in order to run git clone
+        uses: webfactory/ssh-agent@v0.9.0
         with:
-          git-name: Drainpipe Bot
-          git-email: no-reply@example.com
-          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
-          ssh-known-hosts: ${{ secrets.SSH_KNOWN_HOSTS }}
+          ssh-private-key: ${{ secrets.DRAINPIPE_DEV_DEPLOY_KEY }}
+          log-public-key: false
 
-      - name: Setup Project
+      # Commits made by the https://github.com/apps/lullabot-drainpipe application.
+      # @see https://github.com/orgs/community/discussions/24664
+      - name: Create git checkout of drainpipe-dev
+        working-directory: drainpipe-dev
+        run: |
+          git init
+          git branch -m ${{ github.ref_name }}
+          git remote add origin git@github.com:Lullabot/drainpipe-dev.git
+          git fetch origin
+          git reset --mixed origin/${{ github.ref_name }}
+          git config user.name "Lullabot-Drainpipe[bot]"
+          git config user.email "157769597+Lullabot-Drainpipe[bot]@users.noreply.github.com"
+
+      - name: Push branch to drainpipe-dev
+        if: ${{ startsWith(github.ref, 'refs/heads/') }}
+        working-directory: drainpipe-dev
+        run: |
+          git add -A
+          git commit -m "$COMMIT_MESSAGE" --allow-empty
```

Comment:
I wonder if the tag also needs the git commit to be done, to ensure we are pushing the correct code to the tag.